### DISCUSSION TITLE
Units of forces, coordinates and masses

### DISCUSSION THREAD

**Discussion by AnnevanGils** [torchmd/torchmd#46] [Q&A] [answered]
I'm trying to use a custom model for the external forcefield, but I'm wondering what the units are supposed to be of the forces (and potential)? Is this in Newton? Also, I'm providing my own molecule coordinates. Should these also be provided in meters, or can I use Angstroms for these? Same goes for the masses of the coordinates. I'm using the Moleculekit Molecule class. All info is welcome.

**Comment by RaulPPelaez** [comment]
See here for a related discussion:
https://github.com/torchmd/torchmd/pull/43

**Comment by RaulPPelaez** [comment]
@AntonioMirarchi @stefdoerr , is there any mention to units in some TorchMD resource?

**Reply by AntonioMirarchi** [reply, marked as answer]
From [torchmd-paper](https://pubs.acs.org/doi/10.1021/acs.jctc.0c01343) methods: " TorchMD uses chemical units consistent with classical MD codes such as ACEMD, namely kcal/mol for energies, K fortemperatures, g/mol for masses, and Å for distances."

**Reply by AnnevanGils** [reply]
Sorry for the follow up question, but if I'm specifying initial velocities via the system class, should I provide them in Å/s or in sqrt( (kcal/mol) / (g/mol) )?

**Reply by giadefa** [reply]
@stefdoerr it should be internal units right?

**Reply by AnnevanGils** [reply]
Thanks for your replies, I used velocities in sqrt(kcal/g) and that worked for me

**Reply by stefdoerr** [reply]
should be A/s considering that we use it in the integrator as `pos += vel * dt + 0.5 * accel * dt * dt`

**Reply by stefdoerr** [reply]
Although I'm not sure about the maxwell boltzman initialization. @giadefa are the units here correct? I think you wrote that part https://github.com/torchmd/torchmd/blob/master/torchmd/integrator.py#L13-L21

**Comment by giadefa** [comment]
The units are internal units.
